src/System.Management.Automation/DscSupport/JsonDscClassCache.cs

```diff
@@ -0,0 +1,3274 @@
+        /// </summary>
+        internal static readonly Tuple<string, Version> DefaultModuleInfoForMetaConfigResource = new Tuple<string, Version>("PSDesiredStateConfigurationEngine", new Version("2.0"));
+
+        /// <summary>
+        /// A set of dynamic keywords that can be used in both configuration and meta configuration.
+        /// </summary>
+        internal static readonly HashSet<string> SystemResourceNames =
+            new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "Node", "OMI_ConfigurationDocument" };
+
+        /// <summary>
+        /// When this property is set to true, DSC Cache will cache multiple versions of a resource.
+        /// That means it will cache duplicate resource classes (class names for a resource in two different module versions are same).
+        /// NOTE: This property should be set to false for DSC compiler related methods/functionality, such as Import-DscResource,
+        ///       because the Mof serializer does not support deserialization of classes with different versions.
+        /// </summary>
+        [ThreadStatic]
+        private static bool t_cacheResourcesFromMultipleModuleVersions;
+
+        private static bool CacheResourcesFromMultipleModuleVersions
+        {
+            get
+            {
+                return t_cacheResourcesFromMultipleModuleVersions;
+            }
+
+            set
+            {
+                t_cacheResourcesFromMultipleModuleVersions = value;
+            }
+        }
+
+        /// <summary>
+        /// Initialize the class cache with the default classes in $ENV:SystemDirectory\Configuration.
+        /// </summary>
+        public static void Initialize()
+        {
+            Initialize(null, null);
+        }
+
+        /// <summary>
+        /// Initialize the class cache with the default classes in $ENV:SystemDirectory\Configuration.
+        /// </summary>
+        /// <param name="errors">Collection of any errors encountered during initialization.</param>
+        /// <param name="modulePathList">List of module path from where DSC PS modules will be loaded.</param>
+        public static void Initialize(Collection<Exception> errors, List<string> modulePathList)
+        {
+            s_tracer.WriteLine("Initializing DSC class cache force={0}");
+
+            if (Platform.IsLinux || Platform.IsMacOS)
+            {
+                //
+                // Load the base schema files.
+                //
+                ClearCache();
+                var dscConfigurationDirectory = Environment.GetEnvironmentVariable("DSC_HOME") ??
+                                                "/etc/opt/omi/conf/dsc/configuration";
+
+                if (!Directory.Exists(dscConfigurationDirectory))
+                {
+                    throw new DirectoryNotFoundException("Unable to find DSC schema store at " + dscConfigurationDirectory + ". Please ensure PS DSC for Linux is installed.");
```

Comment:
We aren't going to support macOS?

Reply:
We should automatically switch to the resources in the machine module location.

Reply:
Discussed offline. The concern is that `/etc/opt/omi/conf/dsc/configuration` will not exist on Mac. On the other hand, it is used only as backup for `$env:DSC_HOME` which is always set by `PSDesiredStateConfiguration` module during module load. Left this as it is.